sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/OrcReadBenchmark.scala

```diff
@@ -0,0 +1,435 @@
+        ------------------------------------------------------------------------------------------------
+        Native ORC MR (0.0%)                          2534 / 2535          4.1         241.7       1.0X
+        Native ORC Vectorized (0.0%)                   961 /  963         10.9          91.6       2.6X
+        Hive built-in ORC (0.0%)                      3980 / 4005          2.6         379.5       0.6X
+
+        String with Nulls Scan:                  Best/Avg Time(ms)    Rate(M/s)   Per Row(ns)   Relative
+        ------------------------------------------------------------------------------------------------
+        Native ORC MR (0.5%)                          2417 / 2422          4.3         230.5       1.0X
+        Native ORC Vectorized (0.5%)                  1314 / 1315          8.0         125.3       1.8X
+        Hive built-in ORC (0.5%)                      2940 / 2953          3.6         280.4       0.8X
+
+        String with Nulls Scan:                  Best/Avg Time(ms)    Rate(M/s)   Per Row(ns)   Relative
+        ------------------------------------------------------------------------------------------------
+        Native ORC MR (0.95%)                         1306 / 1318          8.0         124.5       1.0X
+        Native ORC Vectorized (0.95%)                  521 /  539         20.1          49.7       2.5X
+        Hive built-in ORC (0.95%)                     1636 / 1641          6.4         156.0       0.8X
+        */
+        benchmark.run()
+      }
+    }
+  }
+
+  def columnsBenchmark(values: Int, width: Int): Unit = {
+    val sqlBenchmark = new Benchmark(s"SQL Single Column Scan from wide table ($width columns)", values)
+
+    withTempPath { dir =>
+      withTempTable("t1", "nativeOrcTable", "hiveOrcTable") {
+        import spark.implicits._
+        val middle = width / 2
+        val selectExpr = (1 to width).map(i => s"value as c$i")
+        spark.range(values).map(_ => Random.nextLong).toDF()
+          .selectExpr(selectExpr: _*).createOrReplaceTempView("t1")
+
+        prepareTable(dir, spark.sql("SELECT * FROM t1"))
+
+        sqlBenchmark.addCase("Native ORC MR") { _ =>
+          withSQLConf(SQLConf.ORC_VECTORIZED_READER_ENABLED.key -> "false") {
+            spark.sql(s"SELECT sum(c$middle) FROM nativeOrcTable").collect()
+          }
+        }
+
+        sqlBenchmark.addCase("Native ORC Vectorized") { _ =>
+          spark.sql(s"SELECT sum(c$middle) FROM nativeOrcTable").collect()
+        }
+
+        sqlBenchmark.addCase("Hive built-in ORC") { _ =>
+          spark.sql(s"SELECT sum(c$middle) FROM hiveOrcTable").collect()
+        }
+
+        /*
+        Java HotSpot(TM) 64-Bit Server VM 1.8.0_152-b16 on Mac OS X 10.13.1
+        Intel(R) Core(TM) i7-4770HQ CPU @ 2.20GHz
+
+        SQL Single Column Scan from wide table (100 columns): Best/Avg Time(ms)    Rate(M/s)   Per Row(ns)   Relative
```

Comment:
nit: `Single Column Scan from wide table`. The current title is too long, the result table is not well aligned.